src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -290,16 +290,15 @@ private List<Process> GetPossibleRunningNodes(out string msbuildName, ref string
                 msbuildLocation = "MSBuild.exe";
             }
 
-
             msbuildName = Path.GetFileNameWithoutExtension(msbuildLocation);
 
-            List<Process> nodeProcesses = new List<Process>(Process.GetProcessesByName(msbuildName));
+            var expectedProcessName = Path.GetFileNameWithoutExtension(GetCurrentHost()) ?? msbuildName;
+
+            List<Process> nodeProcesses = new List<Process>(Process.GetProcessesByName(expectedProcessName));
```

Comment:
In case there's a host, is it worth probing it's commandline to further filter out the non msbuild ones, to avoid the time wasted trying to connect to other processes, like the vbcscompiler? :)

Reply:
I think so -- I'd like to pursue this in a future version. Unfortunately, there's no corefx API for getting another process's command line, so it'll have to be OS-specific.

Reply:
Filed #3125 to follow up on this.